data/mapcss-mapping.csv

```diff
@@ -1014,7 +1014,7 @@ shop|greengrocer;939;
 shop|beverages;940;
 shop|toys;941;
 shop|confectionery;942;
-shop;[shop?];;name;int_name;943;
+shop;[shop?][shop!=vacant];;name;int_name;943;
```

Comment:
Should this type be deleted, and replaced in data/replaced_tags.txt? 
@vng

Reply:
Should add test, but AFAIK != doesn't work

Reply:
> Should add test, but AFAIK != doesn't work

Interesting, I assumed it works after testing it with the generator and desktop app :smile:

Reply:
Ok, great then. Please, add a test in osm_type_test.cpp

Reply:
> > Should add test, but AFAIK != doesn't work
> 
> Interesting, I assumed it works after testing it with the generator and desktop app 😄

I guess that you checked that `shop=vacant` shops are dissappering. But did you check that shops that should not be disappearing are indeed not disappearing? I.e. shops that should be matched by that line and that are not matched by any other line? It happened to me with a similar situation: https://github.com/organicmaps/organicmaps/pull/6174#discussion_r1344691091 I thought that the operator != was working but it wasn't.

Reply:
`!=` doesn't work, this change will make all shops disappear (the line will match nothing) :)

Reply:
I think just add a mapping like
`shop|vacant;1589;`
to catch them.
But don't add a style for this time (you might have to supress/reset the style inherited from a generic `shop`).

Reply:
shop;[shop?][!vacant];;name;int_name;943;
?